Sources/WalletConnectKMS/Serialiser/Serializer.swift

```diff
@@ -22,38 +22,59 @@ public class Serializer {
     /// Encrypts and serializes an object
     /// - Parameters:
     ///   - topic: Topic that is associated with a symetric key for encrypting particular codable object
-    ///   - message: Message to encrypt and serialize
+    ///   - encodable: Object to encrypt and serialize
+    ///   - envelopeType: type of envelope
     /// - Returns: Serialized String
-    public func serialize(topic: String, encodable: Encodable) throws -> String {
+    public func serialize(topic: String, encodable: Encodable, envelopeType: Envelope.EnvelopeType) throws -> String {
         let messageJson = try encodable.json()
-        if let symmetricKey = kms.getSymmetricKeyRepresentable(for: topic) {
-            return try codec.encode(plaintext: messageJson, symmetricKey: symmetricKey)
-        } else {
-            throw Error.symmetricKeyForTopicNotFound
+        guard let symmetricKey = kms.getSymmetricKeyRepresentable(for: topic) else {
+            throw Errors.symmetricKeyForTopicNotFound
         }
+        let sealbox = try codec.encode(plaintext: messageJson, symmetricKey: symmetricKey)
+        return Envelope(type: envelopeType, sealbox: sealbox).serialised()
     }
-    
+
     /// Deserializes and decrypts an object
     /// - Parameters:
     ///   - topic: Topic that is associated with a symetric key for decrypting particular codable object
-    ///   - message: Message to deserialize and decrypt
+    ///   - encodedEnvelope: Envelope to deserialize and decrypt
     /// - Returns: Deserialized object
-    public func tryDeserialize<T: Codable>(topic: String, message: String) -> T? {
+    public func tryDeserialize<T: Codable>(topic: String, encodedEnvelope: String) -> T? {
         do {
-            let deserializedCodable: T
-            if let symmetricKey = kms.getSymmetricKeyRepresentable(for: topic) {
-                return try deserialize(message: message, symmetricKey: symmetricKey)
-            } else {
-                throw Error.symmetricKeyForTopicNotFound
+            let envelope = try Envelope(encodedEnvelope)
```

Comment:
Do we plan to handle errors here?

Reply:
you're right, errors here could be handled better, maybe after next release?